Voycar.Api.Web/Entities/User.cs

```diff
@@ -0,0 +1,25 @@
+namespace Voycar.Api.Web.Entities;
+
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using Generic;
+
+public class User : Entity
+{
+    // Login
+    [EmailAddress]
+    public required string Email{ get; set; }
+    public string PasswordHash { get; set; }
+
+    // Personal Details
+    public required string FirstName { get; set; }
+    public required string LastName { get; set; }
+    public required string Street { get; set; }
+    public required string HouseNumber { get; set; }
+    public required string PostalCode { get; set; }
+    public required string City { get; set; }
+    // Foreign key to Role
+    [ForeignKey("Role")]
+    public Guid RoleId { get; set; }
+    public Role Role { get; set; } // Navigation property, useful for Repository
```

Comment:
Macht es Sinn hier `Role` und `RoleId` zu speichern? Die `RoleId` sollte doch eigentlich aus der `Role` bestimmbar sein

Reply:
Die Role selbst wird nicht in der Datenbank gespeichert, sondern nur die Id. Die Role brauche ich zum Erstellen der Beziehung, um die Abhängigkeiten zwischen der Role und dem User darzustellen. Gleiches beim User und Member. Mit der Role Property (User Property in Member) können wir direkt auf die dazugehörige Role (User bei Member) zugreifen und können uns damit Arbeit sparen.